.github/workflows/daily.yaml

```diff
@@ -68,74 +68,132 @@ jobs:
+              go-yq '
+                .sources[] |
+                select(
+                  . != "https://github.com/truecharts*" and
+                  . != "https://ghcr*" and
+                  . != "docker.io*" and
+                  . != "https://docker.io*" and
+                  . != "https://hub.docker*" and
+                  . != "https://fleet.*" and
+                  . != "https://github.com/truecharts/containers/tree/master/mirror/*" and
+                  . != "https://public.ecr.aws*" and
+                  . != "https://ocir.io*" and
+                  . != "https://gcr*" and
+                  . != "https://azurecr*" and
+                  . != "https://quay*" and
+                  . != "https://lscr*" and
+                  . != "https://github.com/truecharts/containers*" and
+                  . == "http*"
+                )
+                ' \
+              "${chart}/Chart.yaml"
+            )
+            # Empty sources list in-place
+            go-yq -i 'del(.sources.[])' "${chart}/Chart.yaml"
+            # Add truechart source
+            tcsource="https://github.com/truecharts/charts/tree/master/charts/$train/$chartname" go-yq -i '.sources += env(tcsource)' "${chart}/Chart.yaml"
+            # Get the container image name that was parsed out of the Dockerfile for the website.
+            container=$(cat website/docs/charts/description_list.md | grep "\[${chartname}\]" | cut -f3 -d '|' | grep -v 'Not Found' || echo "")
+            # Convert the container image name to a URL.
+            if [ ! -z "$container" ]; then
+              prefix=""
+              case "$container" in
+                lscr.io/linuxserver/*)
+                  prefix="https://fleet.linuxserver.io/image?name="
+                  container=${container#lscr.io/}
+                  ;;
+                tccr.io/truecharts/*)
+                  prefix="https://github.com/truecharts/containers/tree/master/mirror"
+                  container=${container#tccr.io/truecharts/}
+                  ;;
+                mcr.microsoft.com/*)
+                  prefix=""
+                  ;;
+                public.ecr.aws/*)
+                  prefix="https://gallery.ecr.aws/"
+                  container=${container#public.ecr.aws/}
+                  ;;
+                ghcr.io/*)
+                  prefix="https://"
+                  ;;
+                quay.io/*)
+                  prefix="https://"
+                  ;;
+                gcr.io/*)
+                  prefix="https://"
+                  ;;
+                *.azurecr.io/*)
+                  prefix=""
+                  ;;
+                *.ocir.io/*)
+                  prefix=""
+                  ;;
+                # There have been a number of domains used for the Docker Hub registry over the years.
+                # NOTE:  This is also the default case!
+                docker.io/*|index.docker.io/*|registry-1.docker.io/*|registry.hub.docker.com/*|*)
```

Comment:
The default case is a little sneakily hidden here.  I don't think it's awesome but it minimizes repetition without adding a new function.

Reply:
I'm okey with that tbh.